### PR TITLE
Move root `__init__` imports to a new `api` module

### PR DESCRIPTION
**Pull request by gbolmier**
`→` means `depends on`
~~Strikethroughed~~ dependencies have been removed in this PR

- `_bandit` → metrics(base, stats, optim), utils(anomaly(proba, tree, linear_model), compose)
- `anomaly` → base, stats(utils(compose)), proba, tree, linear_model, optim
- `base` →
- `cluster` → base, utils(anomaly(stats), compose)
- `compat` → base, compose, stream(utils(anomaly(stats, proba, tree, linear_model, optim)))
- `compose` → ~~anomaly(stats, proba, tree, linear_model, optim)~~, base, utils
- `datasets` → utils(anomaly(stats, proba, tree, linear_model, optim), compose), stream
- `drift` → base
- `dummy` → base, stats(utils(anomaly(proba, tree, linear_model, optim), compose))
- `ensemble` → base, metrics, stats, tree, drift, utils(anomaly(proba)), linear_model, optim, preprocessing, naive_bayes
- `evaluate` → base, metrics(stats, optim), stream, utils(anomaly(proba, tree, linear_model), compose), datasets
- `facto` → base, optim, utils
- `feature_extraction` → base, stats, utils
- `feature_selection` → base, stats
- `imblearn` → base, linear_model, stats, optim, utils
- `linear_model` → optim, utils(anomaly(stats, proba, tree), compose), base
- `metrics` → base, stats, utils(anomaly(proba, tree, linear_model), compose), optim
- `misc` → stats, utils(anomaly(proba, tree, linear_model, optim), compose)
- `model_selection` → base, compose(anomaly(stats, proba, tree)), linear_model(utils), metrics, optim, preprocessing, _bandit
- `multiclass` → base, linear_model(optim, utils(anomaly(stats, proba, tree), compose))
- `multioutput` → base, linear_model(optim), utils(anomaly(stats, proba, tree), compose)
- `naive_bayes` → base, proba(stats(utils(anomaly(tree, linear_model, optim), compose)))
- `neighbors` → base, utils(anomaly(stats, proba, tree, linear_model, optim), compose)
- `neural_net` → base, optim(utils(anomaly(stats, proba, tree, linear_model), compose))
- `optim` → base, utils(anomaly(stats, proba, tree, linear_model), compose)
- `preprocessing` → base, stats, utils(anomaly(proba, tree, linear_model, optim)), compose
- `proba` → stats(utils(anomaly(tree, linear_model, optim), base, compose))
- `reco` → base, optim, stats, utils(anomaly(proba, tree, linear_model, optim), compose)
- `rules` → anomaly(stats(utils(compose)), proba, optim)), base, drift, linear_model, stats, tree
- `stats` → base, utils(anomaly(proba, tree, linear_model, optim), compose)
- `stream` → utils(anomaly(stats, proba, tree, linear_model, optim), compose), base
- `time_series` → base, metrics(stats, utils(anomaly(proba, tree), compose), optim), linear_model, preprocessing
- `tree` → base, utils(anomaly(stats, proba, tree), compose), linear_model(optim)
- `utils` → ~~anomaly(stats, proba, tree, linear_model, optim)~~, base, ~~compose~~

---

`←` means `is used by`
~~Strikethroughed~~ subpackages are using the dependency indirectly (the dependency is transitive for them)

- `_bandit` ← model_selection
- `anomaly` ← ~~_bandit~~, ~~cluster~~, ~~compat~~, compose, ~~datasets~~, ~~dummy~~, ~~ensemble~~, ~~evaluate~~, ~~linear_model~~, ~~metrics~~, ~~misc~~, ~~model_selection~~, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, ~~neighbors~~, ~~neural_net~~, ~~optim~~, ~~preprocessing~~, ~~proba~~, ~~reco~~, rules, ~~stats~~, ~~stream~~, ~~time_series~~, ~~tree~~, utils
- `base` ← ~~_bandit~~, anomaly, base, cluster, compat, compose, drift, dummy, ensemble, evaluate, facto, feature_extraction, feature_selection, imblearn, linear_model, metrics, model_selection, multiclass, multioutput, naive_bayes, neighbors, neural_net, optim, preprocessing, ~~proba~~, reco, rules, stats, stream, time_series, tree, utils
- `cluster` ←
- `compat` ←
- `compose` ← ~~_bandit~~, ~~anomaly~~, ~~cluster~~, compat, ~~datasets~~, ~~dummy~~, ~~evaluate~~, ~~linear_model~~, ~~metrics~~, ~~misc~~, model_selection, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, ~~neighbors~~, ~~neural_net~~, ~~optim~~, preprocessing, ~~proba~~, ~~reco~~, ~~rules~~, ~~stats~~, ~~stream~~, ~~time_series~~, ~~tree~~, utils
- `datasets` ← evaluate
- `drift` ← ensemble, rules
- `dummy` ←
- `ensemble` ←
- `evaluate` ←
- `facto` ←
- `feature_extraction` ←
- `feature_selection` ←
- `imblearn` ←
- `linear_model` ← ~~_bandit~~, anomaly, ~~compat~~, ~~compose~~, ~~datasets~~, ~~dummy~~, ensemble, ~~evaluate~~, imblearn, metrics, ~~misc~~, model_selection, multiclass, multioutput, ~~naive_bayes~~, ~~neighbors~~, ~~neural_net~~, ~~optim~~, ~~preprocessing~~, ~~proba~~, ~~reco~~, rules, ~~stats~~, ~~stream~~, time_series, tree, ~~utils~~
- `metrics` ← _bandit, ensemble, evaluate, model_selection, time_series
- `misc` ←
- `model_selection` ←
- `multiclass` ←
- `multioutput` ←
- `naive_bayes` ← ensemble
- `neighbors` ←
- `neural_net` ←
- `optim` ← ~~_bandit~~, anomaly, ~~compat~~, ~~compose~~, ~~datasets~~, ~~dummy~~, ensemble, ~~evaluate~~, facto, imblearn, linear_model, metrics, ~~misc~~, model_selection, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, ~~neighbors~~, neural_net, ~~preprocessing~~, ~~proba~~, ~~reco~~, ~~rules~~, ~~stats~~, ~~stream~~, ~~time_series~~, ~~tree~~, ~~utils~~
- `preprocessing` ← ensemble, model_selection, time_series
- `proba` ← ~~_bandit~~, anomaly, ~~compat~~, ~~compose~~, ~~datasets~~, ~~dummy~~, ~~ensemble~~, ~~evaluate~~, ~~linear_model~~, ~~metrics~~, ~~misc~~, ~~model_selection~~, ~~multiclass~~, ~~multioutput~~, naive_bayes, ~~neighbors~~, ~~neural_net~~, ~~optim~~, ~~preprocessing~~, ~~reco~~, ~~rules~~, ~~stats~~, ~~stream~~, ~~time_series~~, ~~tree~~, ~~utils~~
- `reco` ←
- `rules` ←
- `stats` ← ~~_bandit~~, anomaly, ~~cluster~~, ~~compat~~, ~~compose~~, ~~datasets~~, dummy, ensemble, ~~evaluate~~, feature_extraction, feature_selection, imblearn, ~~linear_model~~, metrics, misc, ~~model_selection~~, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, ~~neighbors~~, ~~neural_net~~, ~~optim~~, preprocessing, proba, reco, ~~rules~~, ~~stream~~, ~~time_series~~, ~~tree~~, ~~utils~~
- `stream` ← compat, datasets, evaluate
- `time_series` ←
- `tree` ← ~~_bandit~~, anomaly, ~~compat~~, ~~compose~~, ~~datasets~~, ~~dummy~~, ensemble, ~~evaluate~~, ~~linear_model~~, ~~metrics~~, ~~misc~~, ~~model_selection~~, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, ~~neighbors~~, ~~neural_net~~, ~~optim~~, ~~preprocessing~~, ~~proba~~, ~~reco~~, rules, ~~stats~~, ~~stream~~, ~~time_series~~, ~~utils~~
- `utils` ← _bandit, ~~anomaly~~, cluster, ~~compat~~, compose, datasets, ~~dummy~~, ensemble, evaluate, facto, feature_extraction, feature_selection, imblearn, linear_model, metrics, misc, ~~model_selection~~, ~~multiclass~~, ~~multioutput~~, ~~naive_bayes~~, neighbors, ~~neural_net~~, optim, preprocessing, ~~proba~~, reco, ~~rules~~, stats, stream, ~~time_series~~, tree

### CODE REVIEW

**Comment by MaxHalford** [comment]
@gbolmier I fixed `yamp` :)

**Comment by gbolmier** [comment]
That was quick @MaxHalford! Just re-run the CI and it passed 🚀

File: docs/releases/unreleased.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
 # Unreleased
 
+- Moved all the public modules imports from `river/__init__.py` to `river/api.py` and removed unnecessary dependencies between modules enabling faster cherry-pick imports times (≈ 3X).
```
Comment (MaxHalford) [inline docs/releases/unreleased.md:3]:
```suggestion
- Moved all the public modules imports from `river/__init__.py` to `river/api.py`, and removed unnecessary dependencies between modules enabling faster cherry-pick import times (≈ 3x).
```

File: river/utils/inspect.py [hunk 2]
```diff
@@ -31,13 +33,41 @@ def extract_relevant(model: base.Estimator):
...
+
+    >>> utils.inspect.ischildobject(obj=anomaly.HalfSpaceTrees(), class_name=class_name)
+    True
+
+    >>> utils.inspect.ischildobject(obj=anomaly.OneClassSVM(), class_name=class_name)
+    True
+
+    >>> utils.inspect.ischildobject(obj=anomaly.GaussianScorer(), class_name=class_name)
+    False
+    """
```
Comment (MaxHalford) [inline river/utils/inspect.py:63]:
```suggestion
    False
    
    """
```

**Review by MaxHalford — APPROVED** [review]

**Pull request merged**